### ISSUE TITLE
Extend assignment_linter() to enforce '=' by default for assignment

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2441]
Currently, we don't have a way for enforcing a style guide rule other than `<-`, but not all authors prefer this.

We should extend `assignment_linter()` to allow for a different default operator (with either `<-` or `=` allowed).

**Comment by IndrajeetPatil** [comment]
Agreed. This will also help users who prefer (e.g.) `{formatR}` over `{styler}` as a formatter for R:

``` r
library(formatR)
tidy_source(text = "x = 1")
#> x = 1
```

<sup>Created on 2023-12-14 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
